src/Aspire.Hosting.Azure.AppService/AzureAppServiceWebsiteContext.cs

```diff
@@ -89,6 +91,15 @@ private void ProcessEndpoints()
             throw new NotSupportedException($"The endpoint(s) {string.Join(", ", unsupportedEndpoints.Select(e => $"'{e.Name}'"))} on resource '{resource.Name}' specifies an unsupported scheme. Only http and https are supported in App Service.");
         }
 
+        // App Service supports only one target port
+        var targetPortEndpoints = endpoints.Where(e => e.IsExternal && e.TargetPort is not null).Select(e => e.TargetPort).Distinct();
+        if (targetPortEndpoints.Count() > 1)
+        {
+            throw new NotSupportedException("App Service only supports one target port.");
```

Comment:
This error message isn't great. It should be more detailed.

"App Service does not support resources with multiple external endpoints"